### PR TITLE
Refactored Taproot functions to use `Result` for better error handling

### PR DESCRIPTION
**Pull request by ben2077**
- Updated `tweaked_key`, `tweaked_key_parity`, `tweaked_key_x_only`, `script_path_only`, and `spk` methods to return `Result`.
- Replaced `unwrap()` calls with proper error handling.
- Improved test cases to check for results using `Result`.
- Cleaned up imports for better readability.

This makes Taproot operations more robust and error-proof.

### CODE REVIEW

File: src/taproot.rs [hunk 1]
```diff
@@ -140,11 +138,12 @@ impl TapRoot {
         }
     }
 
-    pub fn script_path_only(branch: Branch) -> TapRoot {
-        TapRoot {
-            inner_key: XOnlyPublicKey::from_slice(&POINT_WITH_UNKNOWN_DISCRETE_LOGARITHM).unwrap(),
```
Comment (brqgoo) [inline src/taproot.rs:145]: It is known with absolute certainty that `XOnlyPublicKey::from_slice` will never fail for our specific hardcoded point, `POINT_WITH_UNKNOWN_DISCRETE_LOGARITHM`. Therefore, it is safe to leave `.unwrap()` as it is.

File: src/taproot.rs [hunk 2]
```diff
@@ -165,37 +164,34 @@ impl TapRoot {
         hash_tap_tweak(&inner_vec, &tweak_vec)
     }
 
-    pub fn tweaked_key(&self) -> PublicKey {
+    pub fn tweaked_key(&self) -> Result<PublicKey, secp256k1::Error> {
         if let Some(_) = &self.uppermost_branch {
-            let scalar: Scalar = Scalar::from_be_bytes(self.tap_tweak()).unwrap();
+            let scalar = Scalar::from_be_bytes(self.tap_tweak()).map_err(|_| secp256k1::Error::InvalidTweak)?;
 
             self.inner_key_lifted()
```
Comment (brqgoo) [inline src/taproot.rs:171]: You forgot to add `Ok()` here as well.

Comment (ben2077) [reply src/taproot.rs:171]: The return value of this function is already wrapped in `result`.

File: src/taproot.rs (hunk 1 above)
Comment (ben2077) [reply src/taproot.rs:145]: I just think that in Rust, we should minimize the use of `unwrap` to maintain a consistent programming style.

**Pull request merged**